src/cadastro_clientes.cpp

```diff
@@ -0,0 +1,98 @@
+#include "cadastro_clientes.hpp"
+#include <algorithm> //remoção de clientes
+#include <regex> //validar cpf e data de nascimento
+
+//FORMATO CPF: xxx.xxx.xxx-xx
+//FORMATO Data de Nascimento: dd/mm/yyyy
+
+//Implementação dos métodos da classe CadastroClientes
+void CadastroClientes::inserirCliente(Cliente* cliente) {
+    //Verificação de existência do cliente
+    if (clienteExiste(cliente->getCPF())) {
+        std::cout << "ERRO: CPF repetido" << std::endl;
+        return;
+    }
+
+    bool DadosIncorretos;
+    //Expressão singular que verifica e valida (ou não) o CPF
+    std::regex regexCPF(R"(\d{3}\.\d{3}\.\d{3}-\d{2})");
+    if (!std::regex_match(cliente->getCPF(), regexCPF)) {
+        DadosIncorretos = true;
+    }
+
+    if (cliente->getNome().empty()){
+        DadosIncorretos = true;
+    }
+    //Expressão singular que verifica e valida (ou não) a data de nascimento
+    std::regex regexDataNascimento(R"(\d{2}/\d{2}/\d{4})");
+    if (!std::regex_match(cliente->getDataNascimento(), regexDataNascimento)) {
+        DadosIncorretos = true;
+    }
+    //Se o CPF estiver inválido, nome vazio ou data de nascimento inválida, retorna erro
+    if (DadosIncorretos) {
+        std::cout << "ERRO: dados incorretos" << std::endl;
+        return;
+    }
+
+    //Adiciona o cliente na lista
+    clientes.push_back(cliente);
+    std::cout << "Cliente de CPF: " << cliente->getCPF() << " cadastrado com sucesso" << std::endl;
+}
+
+void CadastroClientes::removerCliente(const std::string& cpf) {
+    //Procura o cliente na lista
+    auto it = std::find_if(clientes.begin(), clientes.end(), [&cpf](const Cliente* cliente) {
```

Comment:
O auto não é uma boa pratica, afinal gera confução e aumenta o tempo de execução.

Reply:
> O auto não é uma boa pratica, afinal gera confução e aumenta o tempo de execução.

https://www.quora.com/Does-using-auto-make-the-code-slower-in-C

Reply:
Obrigado pela correção.